### ISSUE TITLE
Function App cannot use premium storage

### ISSUE THREAD

**Issue by JohanKroese** [Azure/azure-webjobs-sdk-script#773]
It seems you cannot use an Azure Function App in combination with a blobtrigger (input) or azure table storage (output) in a storage account of type premium.  
#### Repro steps
1. Create storage account, choose 'premium' 
2. Create Azure function, choose type 'blobtrigger'
3. In the log the function will show an error that the storage account is invalid

The error shown:
Microsoft.Azure.WebJobs.Host: Error indexing method 'Functions.fovtestp'. Microsoft.Azure.WebJobs.Host: Invalid storage account 'johanfovtestp'. Please make sure your credentials are correct
1. Create a new storage account (same region, settings, etc) and choose 'standard'
2. switch the storage account connection(s) used in the function to the new storage account. now the function works.
#### Background

Since recently one of my Azure functions is hitting the new 5min timeout on Azure functions (https://github.com/Azure/azure-webjobs-sdk-script/issues/18#issuecomment-254603536). The function writes +/- 200k records to Azure table storage which takes more than 5m. I was wondering how much faster this process would be when I would store the table storage on premium storage. However, the Function App does not seem to support premium storage momentarily. Regardless of whether I use premium storage for the (blob) trigger or for the Azure Table storage output, the function logs the error:

2016-10-18T21:10:16  Welcome, you are now connected to log-streaming service.
2016-10-18T21:10:25.683 Microsoft.Azure.WebJobs.Host: Error indexing method 'Functions.fovtestp'. Microsoft.Azure.WebJobs.Host: Invalid storage account 'johanfovtestp'. Please make sure your credentials are correct.

**Comment by JohanKroese** [comment]
Ok, so I found out that actually premium storage can only be used for blobs, not for tables. who knew :-) 

nevertheless I can image a need for being able to use a blobtrigger on blobs that are stored in premium storage

**Comment by mathewc** [comment]
Our logic for account validation is simple and can be found [here](https://github.com/Azure/azure-webjobs-sdk/blob/master/src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs#L61). Basically we just attempt a simple blob operation to ensure it succeeds. You might try taking the connection string our picker put in your app setting and verifying that it is correct by attempting to do the same validations that we do - create a blob client and call GetServiceProperties.

@mamaso Any ideas?

**Comment by mamaso** [comment]
Looking at premium storage, it appears to be in the same bucket as blob-only storage, meaning no support for tables or queues
https://azure.microsoft.com/en-us/documentation/articles/storage-premium-storage/

@mathewc I had assumed that any storage connections that are not AzureWebJobsStorage/Dashboard created for functions would be secondary, but it seems like that may not be the case.

**Comment by mathewc** [comment]
@JohanKroese What version of our runtime are you using? In the last release we made some improvements to allow these kinds of accounts to be supported. Perhaps you're on an older version, 0.5 or earlier?

@mamaso I agree with you that since he's using a secondary account, it should be going through your new logic.

**Comment by JohanKroese** [comment]
@mathewc I'm using 0.6 (latest), just created the function app. I have tested your suggestion to test the connectionstring; I grabbed the connectionstring from the appsettings and then used the Azure Storage Explorer to explore the storage account, proving that the connectionstring is ok. Interestingly when I try to upload a blob I get:
_Block blobs are not supported._
which suddenly made me realize that this the, very logical, cause of the error: since premium storage is only meant to be used for VHDs and only supports page blobs, using such an account as a blobtrigger makes no sense at all. To be exact: [line 53 in the storage account validation logic](https://github.com/Azure/azure-webjobs-sdk/blob/master/src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs#L53) fails:
`await client.GetServicePropertiesAsync(cancellationToken);`
When I use the validation code on a normal storage account it works fine, if I test it with a premium account it fails with a Microsoft.WindowsAzure.Storage.StorageException telling that the value for one of the query parameters specified in the request URI is invalid. The details in the RequestInformation.ExtendedErrorInformation show that the parameter restype=service is not supported in the request: https://mypremiumstorageaccount.blob.core.windows.net/?comp=properties&restype=service

tl;dr it's probably not very useful to be able to select a premium storage account, since it cannot be used as a blobtrigger and it cannot be used to store tables. I suggest preventing that these accounts can be selected at all.

**Comment by mathewc** [comment]
We do support both Page and Block blobs in BlobTrigger. Perhaps we just need to adjust our account validation check so it works for these types of accounts as well. However, whether it truly makes sense to use premium accounts for BlobTrigger is TBD.

**Comment by paulbatum** [comment]
Closing as a dupe of https://github.com/Azure/azure-webjobs-sdk/issues/810. Reactivate if it turns out there is a script level block on this scenario.

**Comment by lindydonna** [comment]
Reopening per discussion on https://github.com/Azure/azure-webjobs-sdk/issues/810.

**Comment by paulbatum** [comment]
Premium storage now works for input and output scenarios. I filed https://github.com/Azure/azure-webjobs-sdk-script/issues/935 to track the trigger scenario.
